src/main/java/org/folio/rest/impl/LedgerAPI.java

```diff
@@ -149,40 +139,26 @@ private Future<Boolean> isLedgerStatusChanged(Ledger ledger, DBClient client) {
     return promise.future();
   }
 
-  private Future<Void> updateLedger(Ledger ledger, DBClient client) {
-    Promise<Void> promise = Promise.promise();
-
-   client.getPgClient().update(client.getConnection(), LEDGER_TABLE, ledger, "jsonb", " WHERE id='" + ledger.getId() + "'", false, event -> {
-        if (event.failed()) {
-        handleFailure(promise, event);
-      } else {
-        log.info("Ledger record {} was successfully updated", ledger);
-        promise.complete();
-      }
-    });
-    return promise.future();
+  private Future<Void> updateLedger(Ledger ledger, DBConn dbConn) {
+    return dbConn.update(LEDGER_TABLE, ledger, ledger.getId())
+        .recover(ResponseUtils::handleFailure)
```

Comment:
I see in this method handleFaiure is invoked when future failed, and in the next method - not, is it okay?

Reply:
I see that handleFailure invoked in wrapper, so everething is okay